### ISSUE TITLE
Wowhead Artifact Calculator link showing 820 ilvl Relics used

### ISSUE THREAD

**Issue by henryboston** [simulationcraft/simc#2842]
On the HTML results page if you scroll down you will see a link to the Artifact Traits and Relics used in the simulation.

The text is stating that all of the relics are 840 ilvl but when you click the link they are all showing up as 820 ilvl. Same goes for the wowdb links to artifacts. 

http://www.wowhead.com/artifact-calc#44:137412:137472:137365:0:1052:1:1054:1:1057:1:1060:3:1061:5:1063:1:1064:4:1065:3:1066:3:1348:1:

![screen shot 2016-09-16 at 4 41 24 pm](https://cloud.githubusercontent.com/assets/1584876/18602095/73eb63a2-7c2c-11e6-9460-553454e8e027.png)

![screen shot 2016-09-16 at 4 42 09 pm](https://cloud.githubusercontent.com/assets/1584876/18602111/8f86e424-7c2c-11e6-8f63-ae3d293ef7c7.png)

Is this a problem with the links, or did wowhead/wowdb change their relic code, or are the sims actually using 820 relics? I also noticed certain classes are linking 825 relics

**Comment by navv1234** [comment]
Wowhead artifact calculator has no way for us to include the correct ilevel of the relics, because they are done in a slightly complicated way.

The WowDB tooltips for the items show the base ilevel, because I have not yet updated the the tooltip code to include the necessary information.

The simulator internally uses it's own computation based on the input it is given to come at the values it uses. The "relics: { +x ilevels, +x ilevels, +x ilevels } shows the computed values, and in general the "ilevel: x" shows the computed ilevel of the item.

**Comment by henryboston** [comment]
Okay, thank you. I had a sneaking suspicion the links were just not updated but I wanted to make sure that the main sims on the homepage were using 840 :)

So just to clarify the profile line:
`artifact=44:139260:139261:137493:0:1052:1:1054:1:1057:1:1060:3:1061:3:1064:5:1065:3:1066:3:1348:1`

Is for the traits themselves? and `main_hand=the_dreadblades,id=128872,bonus_id=742,gem_id=137412/137472/137365,relic_id=1727/1727/1727`

has the relic IDs?

I'm wanting to sim different relics traits/ilvls, etc.

**Comment by navv1234** [comment]
Yeah, so things are a bit complicated with this artifact stuff. The way we do things is the following:

1) If the artifact line is provided, parse it. If it has relics, then use the artifact line's traits as is
2) If the artifact line is provided but it does not have relics (0s instead), then use the relics provided by the artifact item's relics (gem_id=x/y/z)
3) The artifact weapon's ilevel is always parsed from the gem_id and relic_id options in the weapon.

tl;dr to test different trait setups, just export a fully filled string from the wowhead artifact calculator, and you should be fine.

**Comment by henryboston** [comment]
Awesome, thanks for the quick response!
